src/tests/tribol_mfem_common_plane.cpp

```diff
@@ -0,0 +1,375 @@
+
+    // set initial velocity
+    {
+      mfem::Array<int> attrib_marker(pmesh->attributes.Max());
+      attrib_marker = 0;
+      for (auto move_attrib : move_attribs)
+      {
+        attrib_marker[move_attrib-1] = 1;
+      }
+      mfem::Array<int> vdof_marker(par_fe_space.GetVSize());
+      vdof_marker = 0;
+      for (int e{0}; e < pmesh->GetNE(); ++e)
+      {
+        if (attrib_marker[pmesh->GetElement(e)->GetAttribute()-1])
+        {
+          mfem::Array<int> vdofs;
+          par_fe_space.GetElementDofs(e, vdofs);
+          par_fe_space.DofsToVDofs(2, vdofs);
+          for (auto vdof : vdofs)
+          {
+            vdof_marker[vdof] = 1;
+          }
+        }
+      }
+      for (int i{0}; i < vdof_marker.Size(); ++i)
+      {
+        if (vdof_marker[i])
+        {
+          v[i] = -std::abs(initial_v);
+        }
+      }
+    }
+
+    // recover dirichlet bc tdof list
+    mfem::Array<int> ess_vdof_list;
+    {
+      mfem::Array<int> ess_vdof_marker;
+      mfem::Array<int> ess_bdr(pmesh->bdr_attributes.Max());
+      ess_bdr = 0;
+      for (auto fix_attrib : fix_attribs)
+      {
+        ess_bdr[fix_attrib-1] = 1;
+      }
+      par_fe_space.GetEssentialVDofs(ess_bdr, ess_vdof_marker);
+      mfem::FiniteElementSpace::MarkerToList(ess_vdof_marker, ess_vdof_list);
+    }
+
+    // set up mfem elasticity bilinear form
+    mfem::ConstantCoefficient rho {100.0};
+    mfem::ConstantCoefficient lambda {100000.0};
+    mfem::ConstantCoefficient mu {100000.0};
+    ExplicitMechanics op {par_fe_space, rho, lambda, mu};
+
+    // set up time integrator
+    CentralDiffSolver solver { ess_vdof_list };
+    solver.Init(op);
+
+    // set up tribol
+    tribol::initialize(pmesh->SpaceDimension(), MPI_COMM_WORLD);
+    tribol::registerMfemMesh(
+      0, 0, 1, *pmesh, coords, surf1_attribs, surf2_attribs,
+      tribol::SURFACE_TO_SURFACE, 
+      tribol::NO_CASE, 
+      tribol::COMMON_PLANE, 
+      tribol::FRICTIONLESS,
+      tribol::PENALTY,
+      tribol::BINNING_GRID
+    );
+    tribol::registerMfemVelocity(0, v);
+    tribol::setPenaltyOptions(
+      0,
+      tribol::KINEMATIC,
+      tribol::KINEMATIC_CONSTANT,
+      tribol::NO_RATE_PENALTY
+    );
+    tribol::setKinematicConstantPenalty(0, p_kine);
+    tribol::setKinematicConstantPenalty(1, p_kine);
+
+    int cycle {0};
+    for (double t {0.0}; t < t_end; t+=dt)
+    {
+      tribol::update(cycle, t, dt);
+
+      op.SetTime(t);
+      solver.Step(u, v, t, dt);
+
+      coords += u;
+      if (order == 1)
+      {
+        pmesh->SetVertices(coords);
+      }
+
+      ++cycle;
+    }
+
+    auto local_max = u.Max();
+    max_disp_ = 0.0;
+    MPI_Allreduce(&local_max, &max_disp_, 1, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD);
+  }
+};
+
+TEST_P(MfemCommonPlaneTest, mass_matrix_transfer)
+{
+  EXPECT_LT(std::abs(max_disp_ - 0.000559868), 1.0e-6);
+
+  MPI_Barrier(MPI_COMM_WORLD);
+}
+
+INSTANTIATE_TEST_SUITE_P(tribol, MfemCommonPlaneTest, testing::Values(1));
```

Comment:
Is this test run with order >= 1 and on multiple ranks to test the different code paths (e.g. LOR and in parallel)?